compiler/modules/types/list.bal

```diff
@@ -51,14 +51,9 @@ public function listAtomicTypeAllMemberTypes(ListAtomicType atomicType) returns
     return [ranges, types];
 }
 
-// This is atom index 0
-// Used by bddFixReadOnly
-final ListAtomicType LIST_SUBTYPE_RO = { members: { initial: [], fixedLength: 0 }, rest: READONLY };
-
 public class ListDefinition {
     *Definition;
-    private RecAtom? roRec = ();
-    private RecAtom? rwRec = ();
+    private RecAtom? rec = ();
 
     // The SemType is created lazily so that we have the possibility
     // to share the Bdd between the RO and RW cases.
```

Comment:
I don't think this comment is right any more.  We are creating it lazily so we can only use a recListAtom when necessary.